incubator/hnc/internal/reconcilers/setup.go

```diff
@@ -9,25 +9,57 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 
 	api "sigs.k8s.io/multi-tenancy/incubator/hnc/api/v1alpha2"
 	"sigs.k8s.io/multi-tenancy/incubator/hnc/internal/forest"
 )
 
+// FakeDeleteCRDClient is a "fake" client used for testing only
+type FakeDeleteCRDClient struct{}
+
+// FakeDeleteCRDClient doesn't return any err on Get() because none of the reconciler test performs CRD deletion
+func (f FakeDeleteCRDClient) Get(context.Context, types.NamespacedName, runtime.Object) error {
+	return nil
+}
+
 var crds = map[string]bool{
 	"hncconfigurations.hnc.x-k8s.io":       true,
 	"subnamespaceanchors.hnc.x-k8s.io":     true,
 	"hierarchyconfigurations.hnc.x-k8s.io": true,
 }
 
+// deleteCRDClientType could be either a real client or FakeDeleteCRDClient
+type deleteCRDClientType interface {
+	Get(context.Context, types.NamespacedName, runtime.Object) error
+}
+
+// deleteCRDClient is an uncached client for checking CRD deletion
+var deleteCRDClient deleteCRDClientType
+
 // Create creates all reconcilers.
 //
 // This function is called both from main.go as well as from the integ tests.
-func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int) error {
+func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int, useFakeClient bool) error {
```

Comment:
Since the client is created 100% by variables/info available outside this function, wouldn't it make more sense to just pass the client into this function instead of using a special bool to create a fake client or not?

Reply:
It's a good point, although in this context I'm not sure it's worth the added complexity. The interface and global variable would still have to be declared in this file, while the clients that implement it would now be in cmd/manager/main.go and internal/reconcilers/suite_test.go, so the logic would be fairly scattered. The way @GinnyJI has written it, pretty much everything you need to know is in the same place, and I think that simplicity is worthwhile.

If we start adding a bunch more options here, then I'd be open to refactoring it. wdyt?

Reply:
Yeah, that sounds good.